index.js

```diff
@@ -7,19 +7,43 @@ var analyze = require('./lib/analyze');
 
 /**
  * Unpacks, parses, validates, and analyzes Scratch projects. If successful,
- * will return a valid Scratch project object with appended metadata.
+ * will return a valid Scratch project object with appended metadata, as well
+ * as the unpacked zip object if originally provided.
  *
  * @param {Buffer | string} Input buffer or string representing scratch project
  *
  * @return {Object}
  */
 module.exports = function (input, callback) {
-    async.waterfall([
-        function (cb) {
-            unpack(input, cb);
-        },
-        parse,
-        validate,
-        analyze
-    ], callback);
+    // A wrapper around the analyze function, so that
+    // it only gets called on 2.0 projects
+    var analyzeWrapper = function (validatedProject, cb) {
+        if (validatedProject.projectVersion !== 2) {
+            return cb(null, validatedProject);
+        }
+        analyze(validatedProject, cb);
+    };
+
+    // First unpack the input (need this outside of the async waterfall so that
+    // unoackedProject can be refered to again)
```

Comment:
Fixed the typo!

Good to know about async.auto. I took a look, and tried to take a stab at refactoring to use it but the logic quickly became very complex. I think some of this will be cleaned up when we move 'analyze' out of this library.